x/wasm/internal/keeper/keeper.go

```diff
@@ -170,6 +160,42 @@ func (k Keeper) Execute(ctx sdk.Context, contractAddress sdk.AccAddress, caller
 	return types.CosmosResult(*res), nil
 }
 
+func (k Keeper) Query(ctx sdk.Context, contractAddr sdk.AccAddress, req []byte) (*wasmTypes.QueryResult, sdk.Error) {
```

Comment:
This only does smart queries, right?

I think it would be good to name this QuerySmart or such, and make a QueryRaw that handles the raw queries, even if it is just a few lines: https://github.com/cosmwasm/wasmd/pull/22/files#diff-4abe0cb83913fb6687ff80141ae33310R97-R101

We will expose these to be called from the smart contracts themselves in 0.7, so having them on the keeper is very useful

Reply:
I will return `wasmTypes.QueryResult` in this case from the method. Please let me know if this is a bad idea. :-)

Reply:
In 0.6 go-cosmwasm, it just returns `[]byte`.
So do the simplest thing possible, and I will fix when I merge in 0.6 support